include/voxen/common/world.hpp

```diff
@@ -10,23 +11,33 @@
 namespace voxen
 {
 
+//TODO actual real queue
+struct DebugQueueRtW {
+	glm::dvec3 player_forward_movement_direction{0.0f};
+	glm::dvec3 player_strafe_movement_direction{0.0f};
+	glm::dquat player_orientation = glm::identity<glm::dquat>();
+	double strafe_speed{50};
+	double forward_speed{25};
+};
+
 class World {
 public:
 	World();
 	World(const World &other);
 	~World();
 
 	const Player &player() const noexcept { return m_player; }
+	int64_t tickId() const noexcept {return m_tick_id; }
 
-	double secondsPerTick() const noexcept { return 10.0 / 1000.0; }
+	double secondsPerTick() const noexcept { return 1.0 / 100.0; } // 100 UPS
 
-	void update();
+	void update(DebugQueueRtW& queue, std::chrono::duration<int64_t, std::nano> tick_inverval);
 
 	void walkActiveChunks(std::function<void(const TerrainChunk &)> visitor) const;
-	void render(VulkanRender &render);
-private:
+public:
 	Player m_player;
 	TerrainOctree m_terrain;
+	int64_t m_tick_id;
```

Comment:
I'd suggest changing it to `uint64_t`. In practice there should be no difference (the game will never run for so long to get anywhere close to overflowing it), but in theory signed integer overflow is UB while unsigned one is not. And IMO it is more idiomatic to use unsigned types for versioning.